# Análisis Exploratorio del Titanic

**Curso:** Programación II — Ciencia de Datos  
**Universidad:** Externado de Colombia  
**Integrantes:** Jacobo Beltrán y Helen Rueda

## Introducción

El dataset del Titanic contiene información sobre los pasajeros del famoso
barco que naufragó en 1912. El objetivo de este análisis es explorar los
datos, entender las variables y responder preguntas sobre los pasajeros
y sus probabilidades de supervivencia.

Los datos provienen de Kaggle y están divididos en dos conjuntos:
- **train.csv**: 891 pasajeros con información de supervivencia
- **test.csv**: 418 pasajeros sin información de supervivencia

## 1. Importación de librerías y carga de datos

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use('seaborn-v0_8')
sns.set_palette('husl')

print("Librerias importadas correctamente")
print(f"Version de Pandas: {pd.__version__}")
print(f"Version de NumPy: {np.__version__}")

Librerias importadas correctamente
Version de Pandas: 3.0.3
Version de NumPy: 2.4.6


In [4]:
ruta_train = '/Users/jacobobeltran/Desktop/taller-pandas-titanic-JacoboyHelenF/data/train.csv'
ruta_test = '/Users/jacobobeltran/Desktop/taller-pandas-titanic-JacoboyHelenF/data/test.csv'

df_train = pd.read_csv(ruta_train)
df_test = pd.read_csv(ruta_test)

print("Datasets cargados correctamente")
print(f"Filas en train: {df_train.shape[0]} | Columnas en train: {df_train.shape[1]}")
print(f"Filas en test:  {df_test.shape[0]} | Columnas en test:  {df_test.shape[1]}")

Datasets cargados correctamente
Filas en train: 891 | Columnas en train: 12
Filas en test:  418 | Columnas en test:  11


In [5]:
print("=== Primeras filas de train.csv ===")
df_train.head()

=== Primeras filas de train.csv ===


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [6]:
print("=== Primeras filas de test.csv ===")
df_test.head()

=== Primeras filas de test.csv ===


,PassengerId,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


## 2. Descripción de las variables

El dataset del Titanic contiene las siguientes variables:

| Variable | Tipo | Descripción |
|----------|------|-------------|
| PassengerId | Numérica | Identificador único de cada pasajero |
| Survived | Categórica | Supervivencia (0 = No, 1 = Sí). Solo en train |
| Pclass | Categórica | Clase del tiquete (1 = Primera, 2 = Segunda, 3 = Tercera) |
| Name | Texto | Nombre completo del pasajero |
| Sex | Categórica | Género del pasajero (male/female) |
| Age | Numérica | Edad en años. Puede tener valores faltantes |
| SibSp | Numérica | Número de hermanos o cónyuges a bordo |
| Parch | Numérica | Número de padres o hijos a bordo |
| Ticket | Texto | Número del tiquete |
| Fare | Numérica | Tarifa pagada por el pasajero |
| Cabin | Texto | Número de cabina. Tiene muchos valores faltantes |
| Embarked | Categórica | Puerto de embarque (C = Cherburgo, Q = Queenstown, S = Southampton) |

In [7]:
print("=== Tipos de datos — train ===")
print(df_train.dtypes)
print(f"\nTotal de variables: {df_train.shape[1]}")

=== Tipos de datos — train ===
PassengerId      int64
Survived         int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

Total de variables: 12


In [8]:
print("=== Tipos de datos — test ===")
print(df_test.dtypes)
print(f"\nTotal de variables: {df_test.shape[1]}")
print("\nNota: test no tiene la variable Survived")

=== Tipos de datos — test ===
PassengerId      int64
Pclass           int64
Name               str
Sex                str
Age            float64
SibSp            int64
Parch            int64
Ticket             str
Fare           float64
Cabin              str
Embarked           str
dtype: object

Total de variables: 11

Nota: test no tiene la variable Survived


## 3. Estadísticas básicas y comparación entre datasets

In [9]:
print("=== Estadísticas básicas — train ===")
df_train.describe()

=== Estadísticas básicas — train ===


,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [10]:
print("=== Estadísticas básicas — test ===")
df_test.describe()

=== Estadísticas básicas — test ===


,PassengerId,Pclass,Age,SibSp,Parch,Fare
count,418.000000,418.000000,332.000000,418.000000,418.000000,417.000000
mean,1100.500000,2.265550,30.272590,0.447368,0.392344,35.627188
std,120.810458,0.841838,14.181209,0.896760,0.981429,55.907576
min,892.000000,1.000000,0.170000,0.000000,0.000000,0.000000
25%,996.250000,1.000000,21.000000,0.000000,0.000000,7.895800
50%,1100.500000,3.000000,27.000000,0.000000,0.000000,14.454200
75%,1204.750000,3.000000,39.000000,1.000000,0.000000,31.500000
max,1309.000000,3.000000,76.000000,8.000000,9.000000,512.329200


In [11]:
print("=== Comparación entre datasets ===\n")
print(f"{'':20} {'TRAIN':>10} {'TEST':>10}")
print(f"{'Filas':20} {df_train.shape[0]:>10} {df_test.shape[0]:>10}")
print(f"{'Columnas':20} {df_train.shape[1]:>10} {df_test.shape[1]:>10}")
print(f"{'Edad promedio':20} {df_train['Age'].mean():>10.2f} {df_test['Age'].mean():>10.2f}")
print(f"{'Tarifa promedio':20} {df_train['Fare'].mean():>10.2f} {df_test['Fare'].mean():>10.2f}")
print(f"{'% mujeres':20} {(df_train['Sex']=='female').mean()*100:>10.2f} {(df_test['Sex']=='female').mean()*100:>10.2f}")
print(f"\nNota: train tiene la columna Survived, test no la tiene")

=== Comparación entre datasets ===

                          TRAIN       TEST
Filas                       891        418
Columnas                     12         11
Edad promedio             29.70      30.27
Tarifa promedio           32.20      35.63
% mujeres                 35.24      36.36

Nota: train tiene la columna Survived, test no la tiene


### Comparación entre datasets

Al comparar train y test observamos que:
- **train** tiene 891 pasajeros y **test** tiene 418
- La edad promedio y la tarifa promedio son similares entre ambos datasets
- La proporción de mujeres es comparable entre los dos conjuntos
- Esto sugiere que ambos datasets son representativos de la misma población

## 4. Nueva variable: Familiares

In [12]:
df_train['Familiares'] = df_train['SibSp'] + df_train['Parch']
df_test['Familiares'] = df_test['SibSp'] + df_test['Parch']

print("Variable Familiares creada correctamente")
print(f"\nDistribucion en train:")
print(df_train['Familiares'].value_counts().sort_index())

Variable Familiares creada correctamente

Distribucion en train:
Familiares
0     537
1     161
2     102
3      29
4      15
5      22
6      12
7       6
10      7
Name: count, dtype: int64


### Análisis de la variable Familiares

La variable **Familiares** representa el número total de familiares
que viajaban con cada pasajero a bordo del Titanic.

**Observaciones:**
- La mayoría de los pasajeros viajaban solos (Familiares = 0)
- Los pasajeros con familias pequeñas (1 a 3 familiares) son el
  segundo grupo más grande
- Muy pocos pasajeros viajaban con familias grandes (4 o más)
- Esta variable será útil para analizar si viajar acompañado
  influyó en las probabilidades de supervivencia

## 5. Unión vertical de los datasets

In [13]:
df_train['source'] = 'train'
df_test['source'] = 'test'

df_test['Survived'] = None

print("Columnas en train:", df_train.columns.tolist())
print("Columnas en test: ", df_test.columns.tolist())

Columnas en train: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked', 'Familiares', 'source']
Columnas en test:  ['PassengerId', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked', 'Familiares', 'source', 'Survived']


In [14]:
df_total = pd.concat([df_train, df_test], ignore_index=True)

print("Dataset combinado creado correctamente")
print(f"Filas totales: {df_total.shape[0]}")
print(f"Columnas:      {df_total.shape[1]}")
print(f"\nDistribucion por fuente:")
print(df_total['source'].value_counts())

Dataset combinado creado correctamente
Filas totales: 1309
Columnas:      14

Distribucion por fuente:
source
train    891
test     418
Name: count, dtype: int64


### Decisión sobre la columna Survived en test

El dataset **test** no contiene la variable `Survived` porque
en la competencia de Kaggle es la variable que se debe predecir.

Para poder unir ambos datasets verticalmente sin perder información,
agregamos la columna `Survived` al dataset test con valores `NaN`,
lo que indica que no se conoce la supervivencia de esos pasajeros.

El dataset combinado tiene **1309 pasajeros** en total:
- 891 de train (con información de supervivencia)
- 418 de test (sin información de supervivencia)

## 8. Preguntas concretas sobre los pasajeros

### 8b. Cuantos pasajeros sobrevivieron y cuantos murieron

In [16]:
# Conteo de sobrevivientes y fallecidos usando df_train
sobrevivieron = df_train[df_train['Survived'] == 1].shape[0]
murieron = df_train[df_train['Survived'] == 0].shape[0]
total = df_train.shape[0]

print("=== Supervivencia de pasajeros ===")
print(f"Sobrevivieron: {sobrevivieron} pasajeros ({sobrevivieron/total*100:.2f}%)")
print(f"Murieron:      {murieron} pasajeros ({murieron/total*100:.2f}%)")
print(f"Total:         {total} pasajeros")

=== Supervivencia de pasajeros ===
Sobrevivieron: 342 pasajeros (38.38%)
Murieron:      549 pasajeros (61.62%)
Total:         891 pasajeros


### Observacion

De los 891 pasajeros del dataset train, menos de la mitad sobrevivio
al naufragio. La tasa de mortalidad fue significativamente alta, lo que
refleja las condiciones del desastre y la insuficiencia de botes salvavidas
para todos los pasajeros a bordo.

### 8d. Cuantos pasajeros viajaron con un familiar a bordo

In [17]:
# Pasajeros que viajaron con al menos un familiar
con_familiar = df_train[df_train['Familiares'] > 0].shape[0]
sin_familiar = df_train[df_train['Familiares'] == 0].shape[0]
total = df_train.shape[0]

print("=== Pasajeros con y sin familiares ===")
print(f"Con familiar:  {con_familiar} pasajeros ({con_familiar/total*100:.2f}%)")
print(f"Sin familiar:  {sin_familiar} pasajeros ({sin_familiar/total*100:.2f}%)")
print(f"Total:         {total} pasajeros")

=== Pasajeros con y sin familiares ===
Con familiar:  354 pasajeros (39.73%)
Sin familiar:  537 pasajeros (60.27%)
Total:         891 pasajeros


### Observacion

Una parte significativa de los pasajeros viajaba acompanada de al menos
un familiar a bordo. Esto es relevante porque mas adelante analizaremos
si viajar acompanado influyo en las probabilidades de supervivencia.

### 9b. Sobrevivieron mas ninos que hombres adultos proporcionalmente

Creemos que si, los ninos tuvieron mayor tasa de supervivencia que los
hombres adultos porque en situaciones de emergencia se prioriza evacuar
primero a los ninos.

In [18]:
# Filtramos ninos y hombres adultos del dataset train
ninos = df_train[df_train['Age'] < 18]
hombres_adultos = df_train[(df_train['Age'] >= 18) & (df_train['Sex'] == 'male')]

tasa_ninos = ninos['Survived'].mean() * 100
tasa_hombres = hombres_adultos['Survived'].mean() * 100

print("=== Supervivencia: ninos vs hombres adultos ===")
print(f"Ninos (menores de 18):")
print(f"  Total:         {len(ninos)}")
print(f"  Sobrevivieron: {ninos['Survived'].sum()}")
print(f"  Tasa:          {tasa_ninos:.2f}%")

print(f"\nHombres adultos (18 o mas):")
print(f"  Total:         {len(hombres_adultos)}")
print(f"  Sobrevivieron: {hombres_adultos['Survived'].sum()}")
print(f"  Tasa:          {tasa_hombres:.2f}%")

=== Supervivencia: ninos vs hombres adultos ===
Ninos (menores de 18):
  Total:         113
  Sobrevivieron: 61
  Tasa:          53.98%

Hombres adultos (18 o mas):
  Total:         395
  Sobrevivieron: 70
  Tasa:          17.72%


### Observacion

Los datos confirman que los ninos tuvieron una tasa de supervivencia
significativamente mayor que los hombres adultos, lo que es consistente
con la politica de evacuacion que priorizaba mujeres y ninos.

### 9d. Los sobrevivientes partieron mas de Southampton que de otros puertos proporcionalmente

Creemos que no necesariamente, ya que aunque Southampton tenia mas pasajeros
en total, eso no implica que tuvieran mayor tasa de supervivencia. La clase
del tiquete y el sexo probablemente influyeron mas que el puerto de embarque.

In [19]:
# Tasa de supervivencia por puerto de embarque
supervivencia_puerto = df_train.groupby('Embarked')['Survived'].agg(['sum', 'count', 'mean'])
supervivencia_puerto.columns = ['Sobrevivieron', 'Total', 'Tasa']
supervivencia_puerto['Tasa'] = (supervivencia_puerto['Tasa'] * 100).round(2)

print("=== Supervivencia por puerto de embarque ===")
print(supervivencia_puerto.to_string())

=== Supervivencia por puerto de embarque ===
          Sobrevivieron  Total   Tasa
Embarked                             
C                    93    168  55.36
Q                    30     77  38.96
S                   217    644  33.70


### Observacion

Los datos muestran que los pasajeros embarcados en Cherbourg tuvieron
una tasa de supervivencia mayor que los de Southampton y Queenstown.
Esto posiblemente se debe a que Cherbourg tenia una mayor proporcion
de pasajeros de primera clase, lo que influyo en sus probabilidades
de supervivencia.

## 10. Grupos familiares y distribucion de cabinas

In [20]:
# Clasificacion de pasajeros en grupos familiares
def clasificar_familia(familiares):
    if familiares == 0:
        return 'Grupo 1: Sin familiares'
    elif familiares <= 3:
        return 'Grupo 2: Familia pequena'
    else:
        return 'Grupo 3: Familia grande'

df_train['grupo_familiar'] = df_train['Familiares'].apply(clasificar_familia)
df_test['grupo_familiar'] = df_test['Familiares'].apply(clasificar_familia)

# Distribucion de cabinas por grupo familiar
df_train['tiene_cabina'] = df_train['Cabin'].notna()

distribucion = df_train.groupby('grupo_familiar')['tiene_cabina'].agg(['sum', 'count', 'mean'])
distribucion.columns = ['Con cabina', 'Total', 'Proporcion']
distribucion['Proporcion'] = (distribucion['Proporcion'] * 100).round(2)

print("=== Distribucion de cabinas por grupo familiar ===")
print(distribucion.to_string())

=== Distribucion de cabinas por grupo familiar ===
                          Con cabina  Total  Proporcion
grupo_familiar                                         
Grupo 1: Sin familiares           94    537       17.50
Grupo 2: Familia pequena         104    292       35.62
Grupo 3: Familia grande            6     62        9.68


### Observacion

Los pasajeros sin familiares a bordo representan el grupo mas grande.
Los pasajeros con cabina asignada son mayoritariamente de primera clase,
independientemente del grupo familiar. Las familias grandes tienen muy
poca presencia de cabinas asignadas, lo que sugiere que viajaban
principalmente en tercera clase.